### PR TITLE
snake camel removal

### PR DESCRIPTION
**Pull request by mihaidma**
Removed snake camel conversion, added mapper functions that enable the user to define its own to column name and from column name conversion.

### CODE REVIEW

File: lib/query-builder.js [hunk 1]
```diff
@@ -2,54 +2,70 @@
...
 var _ = require('lodash')
 var OpParser = require('./operator_parser')
-var StandardQuery = require('seneca-standard-query')
 
-function parseExtendedExpression (entp, sTypes, current_name, current_value, params, values) {
-  function parseComplexSelectOperator (sTypes, current_name, current_value, params, values) {
-    var result = {
-      processed: false
-    }
+module.exports = function (seneca, opts) {
```
Comment (floridemai) [inline lib/query-builder.js]: What is opts used for? :alien:

File: lib/query-builder.js [hunk 2]
```diff
@@ -2,54 +2,70 @@
...
-        result.err = err
-        return result
+
+      for (var op in currentValue) {
+        var op_val = currentValue[op]
+        if (!OpParser[op]) {
+          result.err = 'This operator is not yet implemented: ' + op
+          return result
+        }
+        var err = OpParser[op](currentName, op_val, params, values, sTypes)
```
Comment (floridemai) [inline lib/query-builder.js]: `opVal`?

File: lib/query-builder.js [hunk 3]
```diff
@@ -2,54 +2,70 @@
...
+        if (err) {
+          result.err = err
+          return result
+        }
+
+        result.processed = true
```
Comment (floridemai) [inline lib/query-builder.js]: Is this working ok? If the first operator is ok (processed=true) and the second one fails you return `processed:true`.

How about:

```
      var result = {}

      result.processed = _.every(currentValue, function (opVal, op) {
        if (!OpParser[op]) {
          result.err = 'This operator is not yet implemented: ' + op
          return false
        }

        var err = OpParser[op](currentName, opVal, params, values, sTypes)
        if (err) {
          result.err = err
          return false
        }

        return true
      })

      return result
```

File: lib/query-builder.js [hunk 4]
```diff
@@ -2,54 +2,70 @@
...
       }
 
-      result.processed = true
+      return result
+    }
+
+    var result = StandardQuery.parseExpression(sTypes, currentName, currentValue, params, values)
+
+    if (!result.processed) {
```
Comment (floridemai) [inline lib/query-builder.js]: Combine ifs? `if (!result.processed && _.isObject(currentValue) {`

File: store-query.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
 'use strict'
 
 var _ = require('lodash')
-var QueryBuilder = require('./lib/query-builder')
-
 var actionRole = 'sql'
```
Comment (floridemai) [inline store-query.js:4]: I know it is not part of the PR, but the name is super-generic. I'm curios how it is used? It is used only internally?

File: test/store.test.js [hunk 6]
```diff
@@ -19,8 +19,10 @@ var si = Seneca({
 describe('seneca-store-query tests ', function () {
   before({}, function (done) {
     si.use(require('seneca-postgres-store'), DBConfig)
-    si.use(require('..'))
-    si.ready(done)
+    si.ready(function () {
```
Comment (floridemai) [inline test/store.test.js:22]: :clap:

File: lib/query-builder.js [hunk 7]
```diff
@@ -2,54 +2,70 @@
...
+          return result
+        }
+      }
     }
 
     return result
   }
 
-  var result = StandardQuery.parseExpression(entp, sTypes, current_name, current_value, params, values)
+  function buildQueryFromExpression (entp, query_parameters, sTypes, values) {
```
Comment (floridemai) [inline lib/query-builder.js]: `query_parameters`? :stuck_out_tongue:

File: lib/query-builder.js (hunk 3 above)
Comment (mihaidma) [inline lib/query-builder.js]: :+1:

**Comment by floridemai** [comment]
:ship: IT

**Pull request merged**